File: WebServices_lab/src/main/java/org/example/component/column/ComplexIntegerColumn.java
package org.example.component.column;

import org.example.component.Column;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ComplexIntegerColumn extends Column {

  public ComplexIntegerColumn(String name) {
    super(name);
    this.type = ColumnType.COMPLEXINT.name();
  }

  @Override
  public boolean validate(String data) {
    // Define a regular expression pattern for complex numbers
    String complexIntegerPattern = "^(0|[-]?[1-9]\\d*)([-+](0|[1-9]\\d*))i$";
    Pattern pattern = Pattern.compile(complexIntegerPattern);
    Matcher matcher = pattern.matcher(data);

    // Check if the input data matches the pattern
    if (matcher.matches() || data.isEmpty()) {
      // If it matches, it's a valid complex number
      return true;
    } else {
      // If it doesn't match, it's not a valid complex number
      return false;
    }
  }
}

File: WebApplication_lab/Classes/ComplexIntegerColumn.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;
using System.Web;

namespace WebApplication.Classes
{
    public class ComplexIntegerColumn : Column
    {
        public ComplexIntegerColumn(string name) : base(name)
        {
            this.Type = ColumnType.COMPLEXINT.ToString();
        }

        public override bool Validate(string data)
        {
            // Define a regular expression pattern for complex numbers
            string complexIntegerPattern = @"^(0|[-]?[1-9]\d*)([-+](0|[1-9]\d*))i$";
            Regex regex = new Regex(complexIntegerPattern);
            Match match = regex.Match(data);

            // Check if the input data matches the pattern
            return match.Success || string.IsNullOrEmpty(data);
        }
    }
}
